docs/community/training/debugging/example-python-vs-r.md

```diff
@@ -0,0 +1,112 @@
+# Example: Python vs R
+
+=== "Overview"
+
+    Here we have provided SIR ODE model implementations in Python and in R.
+    Each script runs several scenarios and produces a plot of infection prevalence for each scenario.
+
+    You can download each script to debug on your computer:
+
+    - [sir_ode.py](sir_ode.py)
+    - [sir_ode.R](sir_ode.R)
+
+=== "Python"
+
+    ```py title="sir_ode.py" linenums="1"
+    --8<-- "sir_ode.py"
+    ```
+
+=== "R"
+
+    ```R title="sir_ode.R" linenums="1"
+    --8<-- "sir_ode.R"
+    ```
+
+??? bug "The model outputs differ!"
+
+    Here are prevalence time-series plots produced by each script:
+
+    === "Python plot"
+
+        <figure markdown="span">
+          ![Python outputs](sir_ode_python.png)
+          <figcaption>Model outputs for the Python script.</figcaption>
+        </figure>
+
+    === "R plot"
+
+        <figure markdown="span">
+          ![R outputs](sir_ode_r.png)
+          <figcaption>Model outputs for the R script.</figcaption>
+        </figure>
+
+??? note "Some initial thoughts ..."
+
+    - Is it obvious whether one of the figures is correct and the other is wrong?
+
+    - The `sir_rhs()` functions in the two scripts appear to be equivalent — but are they?
+
+    - The `default_settings()` functions appear to be equivalent — but are they?
+
+    - The `run_model_scaled_beta()` and `run_model_scaled_gamma()` functions also appear to be equivalent.
+
+    - Where might you begin looking?
+
+??? info "Commentary"
+
+    > I've heard that dicts (and other mutable data types) are passed by reference, as such a function could change the “local” value of a variable, which causes an error when using the “global” variable in another function.
+    >
+    > This would also be good for showing that your test cases aren’t perfect.
+    >
+    > You can run function 1 tests and get good answers, and then run function 2 test and get good answers, but this is because they are not sharing the state of the mutable variable.
+    > Sounds like a great inclusion as a mini chapter.
+
+    These are great ideas! The one drawback of the mutable data example is that it won't work in R, because it's copy-on-write. But that's no big deal. In fact, now that I think about, this difference us surely worth highlighting! People who are familiar with only one of these two languages need to be aware of this important difference if they begin using the other language.
```

Comment:
This is great. I think it will shock some people to realise that their two favourite languages have drastic different behaviour.

Reply:
Awesome, thanks! I'm not sure how much (if any) of the "Commentary" box I should leave in the published materials ... perhaps I should collect explanations for the two exercises in a separate page at the end?